app/src/main/java/com/github/dedis/student20_pop/ui/HomeFragment.java

```diff
@@ -88,41 +94,33 @@ public long getItemId(int position) {
 
         @Override
         public View getView(int position, View convertView, ViewGroup parent) {
+            PoPApplication app = (PoPApplication)(getActivity().getApplication());
             if (convertView == null) {
                 LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
                 convertView = inflater.inflate(R.layout.layout_lao_home, null);
             }
             Lao lao = laos.get(position);
             ((TextView) convertView.findViewById(R.id.lao_name)).setText(lao.getName());
             ((TextView) convertView.findViewById(R.id.date)).setText("last interacted with ...");
-            boolean isOrganizer = lao.getOrganizer().equals(myPublicKey);
+            boolean isOrganizer = lao.getOrganizer().equals(id);
             if (isOrganizer){
                 ((TextView) convertView.findViewById(R.id.role)).setText(R.string.organizer);
             }else {
                 ((TextView) convertView.findViewById(R.id.role)).setText(R.string.attendee);
             }
             convertView.setOnClickListener(clicked -> {
                if (isOrganizer){
-                   startOrganizerUI(lao);
+                   startOrganizerUI(app, lao);
                }else {
-                   startAttendeeUI(lao);
+                   startAttendeeUI(app, lao);
```

Comment:
you can replace these line by : 
`app.setCurrentLao(lao);
Intent intent = new Intent(this.getActivity(), isOrganizer ? OrganzierActivity.class : AttendeeActivity.class);
startActivity(intent);
`
It avoids creating two very similar private functions 
(But this really a detail)

Reply:
done